pyflow/RCDT2/Nodes/core.py

```diff
@@ -0,0 +1,183 @@
+
+    def execute_parallel(self, *_args: any, **_kwargs: any) -> None:
+        if self.parallel_executor_is_active:
+            logger.error("Executor is already working. Rejected.")
+            return
+        thread = Thread(target=self.execution_steps)
+        thread.start()
+
+    def execution_steps(self) -> None:
+        self.parallel_executor_is_active = True
+        self.execute()
+        self.parallel_executor_is_active = False
+
+
+class Service(PyflowParallelExecutor):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.service_type: type
+
+        self.createInputPin("In", "ExecPin", callback=self.execute_parallel)
+        self.exec_out: PinBase = self.createOutputPin("Out", "ExecPin")
+
+        for service_part in ["Request", "Response"]:
+            self.create_data_pins(service_part)
+
+        self.topic_name = underscore(self.service_type.__name__)
+        self.client = PyflowRosBridge.node.create_client(
+            self.service_type, self.topic_name
+        )
+
+    def execute(self, *_args: any, **_kwargs: any) -> None:
+        request = self.create_request_from_pins()
+        response = self.call_service(request)
+        if not response.success:
+            logger.error(
+                f"Call to {self.topic_name} with type {self.service_type.__name__} was unsuccessfull."
+            )
+            return
+        logger.info("Response was succesfull.")
+        self.set_response_to_pins(response)
+        self.exec_out.call()
+
+    def create_data_pins(self, service_part: str) -> None:
+        msg: type = getattr(self.service_type, service_part)
+        fields_and_field_types: dict[str, str] = msg.get_fields_and_field_types()
+        for field, field_type in fields_and_field_types.items():
+            pin_type = get_pin_type(field_type)
+            if msg.__name__ == self.service_type.__name__ + "_Request":
+                pin = self.createInputPin(field, pin_type)
+                self.pin_manager.add_input_pin(field, pin)
+            elif msg.__name__ == self.service_type.__name__ + "_Response":
+                pin = self.createOutputPin(field, pin_type)
+                self.pin_manager.add_output_pin(field, pin)
+            else:
+                logger.warning("Message is not a Request or a Response. Skip.")
+
+    def create_request_from_pins(self) -> object:
+        request = self.service_type.Request()
+        for pin_name in self.pin_manager.get_input_pin_names():
+            data = self.pin_manager.get_data(pin_name)
+            setattr(request, pin_name, data)
+        return request
+
+    def set_response_to_pins(self, response: object) -> None:
+        for pin_name in self.pin_manager.get_output_pin_names():
+            data = getattr(response, pin_name)
+            self.pin_manager.set_data(pin_name, data)
+
+    def call_service(self, request: object) -> object:
+        logger.info("Calling service...")
```

Comment:
Hier zouden we nog de naam van de service kunnen aangeven (ik gebruik hier als service_name ipv topic_name):

```suggestion
        logger.info(f"Calling service {self.service_name}...")
```